src/cluster.c

```diff
@@ -5837,11 +6066,22 @@ clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, in
                 /* If it is not the first key, make sure it is exactly
                  * the same key as the first we saw. */
                 if (!equalStringObjects(firstkey,thiskey)) {
-                    if (slot != thisslot) {
+                    /* If the channel exists across different slots, it is still
```

Comment:
if might be easier to follow this code changing the if/else chain to be:
```
if (firstkey == NULL) {
    ... Handle first key
} else if (!is_pubsub_local) {
    ... Check it's really the same key
} else {
    ... Check same node, since pubsublocal
}
```
It's a little duplication, but I think it'll help with readability.

Reply:
Sure. Updated.